CodeU_Assignement_1/src/codeu_assignement_1/SinglyLinkedList.java

```diff
@@ -0,0 +1,236 @@
+            index -= 1; // i = i-1; to get the item befoer where we add
+            for (int i = 0; i < size; i++) 
+            {
+                if (i == index) 
+                {
+                    // Before: curr -> next_curr
+                    Node next_curr = curr.getNext() ;
+                    curr.setNext(n);
+                    n.setNext(next_curr);
+                    break;
+                    // After: curr -> n -> next_curr
+                }
+
+                // Go to next
+                curr = curr.getNext();
+            }
+        }
+        
+        size++ ;
+    }
+    
+    /*
+        Add all the items in the collection into the list
+        NullPointerException - if the specified collection is null
+    */
+    public boolean addAll(Collection<? extends E> c)
+    {
+        if (c == null)
+            throw new NullPointerException();
+        
+        Iterator <? extends E> itr = c.iterator();
+        if (!itr.hasNext()) 
+            return false;
+        
+        // If there are items - add them one after another
+        while (itr.hasNext()) 
+        {
+            this.addLast(itr.next());
+        }
+        
+        return (size > 0); // Shall add something
+    }
+    
+    /*  Add e element in the front/begining of the list  */
+    public E removeFirst() { return removeAt(0); }
+    
+    /*  Add e element in the tail/end of the list  */
+    public E removeLast() { return removeAt(size-1); }
+    
+    /*  Add e element at index  */
+    public E removeAt(int index)
+    {        
+        // check - index is begger than size or neg.
+        if (index < 0 || index >= size)
+            throw new IndexOutOfBoundsException();
+        
+        if (index == 0) // Remove first 
+        { 
+            E ret = (E) m_first.getData();
+            m_first = m_first.getNext();
+            size--;
+            return ret;
+        }
+        else if (index == (size -1)) // Remove last
+        {
+            // Find prev to the last
+            Node prev = m_first;
+            Node curr = m_first;
+            while (curr.getNext() != null)
+            {
+                prev = curr;
+                curr = curr.getNext();
+            }
+            
+            // Return curr, and remove it from the list
+            m_last = prev;
+            m_last.setNext(null);
+            size --;
+            
+            return (E) curr.getData();
+        } 
+        else // The general case where 0 < index < (size-1) 
+        {
+            Node prev = m_first;
+            index -= 1 ; // i = i-1; to get the item befoer where we remove
+            for (int i = 0; i < size - 2; i++) 
+            {
+                if (i == index) 
+                {
+                    // prev -> curr -> next
+                    Node curr = prev.getNext();
+                    Node next = curr.getNext();
+                    prev.setNext(next);
+                    curr.setNext(null); // gc 

```

Comment:
To be clearer, you only clear the next field of the node being removed in this case. If you wanted to be consistent, you'd do this in the other cases where the next field is not already null.